ivy/functional/ivy/layers.py

```diff
@@ -115,31 +115,204 @@ def dropout(x, prob, scale=True, out: Optional[ivy.Array] = None):
 # Attention #
 
 
-@to_native_arrays_and_back
 def scaled_dot_product_attention(
-    q, k, v, scale, mask=None, out: Optional[ivy.Array] = None
-):
+    q: Union[ivy.Array, ivy.NativeArray], 
+    k: Union[ivy.Array, ivy.NativeArray],
+    v: Union[ivy.Array, ivy.NativeArray], 
+    scale: float, 
+    mask: Optional[Union[ivy.Array, ivy.NativeArray]]=None) -> Union[ivy.Array, ivy.NativeArray]:
     """Applies scaled dot product attention to inputs x using optional mask.
-
     Parameters
     ----------
     q
-        The queries *[batch_shape,num_queries,feat_dim]*.
+        The queries input array. The shape of queries input array should be in *[batch_shape,num_queries,feat_dim]*. 
+        The queries input array should have the same size as keys and values.
     k
-        The keys *[batch_shape,num_keys,feat_dim]*.
+        The keys input array. The shape of keys input array should be in *[batch_shape,num_keys,feat_dim]*. 
+        The keys input array should have the same size as queries and values.
     v
-        The values *[batch_shape,num_keys,feat_dim]*.
+        The values input array. The shape of values input should be in *[batch_shape,num_keys,feat_dim]*.
+        The values input array should have the same size as queries and keys.
     scale
-        The value by which to scale the query-key pairs before softmax.
+        The scale float value.
+        The scale float value is used to scale the query-key pairs before softmax.
     mask
-        The mask to apply to the query-key values. Default is None.
-        *[batch_shape,num_queries,num_keys]*
+        The mask input array. The mask to apply to the query-key values. Default is None.
+        The shape of mask input should be in *[batch_shape,num_queries,num_keys]*.
 
     Returns
     -------
     ret
-        The output following application of scaled dot-product attention.
-        *[batch_shape,num_queries,feat_dim]*
+        The output following application of scaled dot-product attention. 
+        The output array is the weighted sum produced by the attention score and value.
+        The shape of output array is *[batch_shape,num_queries,feat_dim]* .
+
+    
```

Comment:
Thanks for the changes! You might not have noticed one of my previous messages so I'll repost it here:

Since ivy.Container inputs can be used with this function, please copy the following lines of text and paste it into Line 150:

Both the description and the type hints above assumes an array input for simplicity,
but this function is *nestable*, and therefore also accepts :code:`ivy.Container`
instances in place of any of the arguments.

https://github.com/unifyai/ivy/blob/44023d5cb58c05c28861b16866b346107f61f02f/ivy/functional/ivy/elementwise.py#L547

Reply:
Added this line :)

Reply:
@chie2727